core/sql/sqlcomp/CmpSeabaseDDLupgrade.cpp

```diff
@@ -1657,7 +1717,7 @@ short CmpSeabaseMDupgrade::executeSeabaseMDupgrade(CmpDDLwithStatusInfo *mdui,
 		    {
 		      *CmpCommon::diags() << DgSqlCode(-20123);
 
-		      mdui->setStep(UPGRADE_FAILED_RESTORE_OLD_REPOS);
+		      mdui->setStep(UPGRADE_FAILED_RESTORE_OLD_LIBRARIES);
```

Comment:
We have already completed the upgrade library step, should this step (and others that follow) indicate we were in the update version step?  In a subsequent release, we may not be upgrading libraries but something else.

Reply:
We ar ein the phase of rewinding the upgrade. Unless the "needsUpgrade" flag is set, the restore steps wont do anything. So we are ok. We just need to start with last step and rewind but some of the in between steps may not do any rewinding.